### PR TITLE
Add clang format

### PR DESCRIPTION
**Pull request by GretaCB**
Per https://github.com/mapbox/hpp-skel/issues/31

### Next Actions
- [x] Add `.clang-format`
- [x] Add `./scripts/format.sh`
- [x] Run `./scripts/format.sh` against hpp-skel code
- [x] Add clang-format Travis build
- [x] Add CONTRIBUTING.md
- [x] Update `.gitignore` per installing clang-format
- [x] Update CHANGELOG.md
- [x] Code Review
- [x] Merge after Travis is 🍏 

cc @springmeyer @mapsam

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -23,6 +23,16 @@ matrix:
...
           sources: [ 'ubuntu-toolchain-r-test' ]
           packages: [ 'libstdc++6', 'libstdc++-5-dev' ]
+    # Clang format build
+    - os: linux
+      language: generic
+      env: BUILDTYPE=debug
+      node_js: 4
+      # Overrides `install` to avoid initializing clang toolchain
+      install: 
+        - ./scripts/format.sh
```
Comment (springmeyer) [inline .travis.yml]: Let's add `make format` target and call that here.

Comment (springmeyer) [reply .travis.yml]: Oh, also looks like you copied from `node-cpp-skel` matrix. So things are a little different here in how to override. Follow my example at https://github.com/mapbox/hpp-skel/pull/42/files#diff-354f30a63fb0907d4ad57269548329e3R5 as you just need to override `script` and you don't need to specify `node_js` or `env` or `language`.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Developer
+
+We use [`clang-format`](https://clang.llvm.org/docs/ClangFormat.html) version [4.0.0](https://github.com/mapbox/mason/tree/master/scripts/clang-format) to consistently format the code base. The format is automatically checked via a Travis CI build as well. Run the following script locally to ensure formatting is ready to merge:
```
Comment (springmeyer) [inline CONTRIBUTING.md]: We actually use version `4.0.1` currently. How about we remove the version mention here and instead say "Look inside https://github.com/mapbox/hpp-skel/blob/master/scripts/setup.sh#L7` to see the version in use.

File: scripts/format.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+Runs clang-format on the code in include/
+
+Return `1` if there are files to be formatted, and automatically formats them.
+
+Returns `0` if everything looks properly formatted.
+
+'
+ # Set up the environment by installing mason and clang++
+ # See https://github.com/mapbox/node-cpp-skel/blob/master/docs/extended-tour.md#configuration-files
```
Comment (springmeyer) [inline scripts/format.sh]: Remove this node-cpp-skel link?

**Review by springmeyer — CHANGES_REQUESTED** [review]
Looking great. Made some minor comments about things needing fixed above.

**Comment by GretaCB** [comment]
Thanks so much for the thorough CR @springmeyer 👍  Implemented and waiting on Travis

**Review by springmeyer — APPROVED** [review]

**Pull request merged**